src/Controls/src/Core/Window/Window.cs

```diff
@@ -679,6 +679,11 @@ public NavigationImpl(Window owner)
 				_owner = owner;
 			}
 
+			protected override Task OnPushAsync(Page page, bool animated)
+			{
+				throw new InvalidOperationException("Page must be wrapped into a navigation page to perform navigation");
```

Comment:
can we add this to all relevant overrides?

Notice how in XF we threw this on pretty much every override except the modal ones
https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Platform.iOS/Platform.cs#L143

Reply:
Done :)